app/components/post_draft/draft_input/index.tsx

```diff
@@ -164,13 +164,16 @@ function DraftInput({
         postPriority,
     });
 
-    const handleSendMessage = useCallback(async () => {
+    const handleSendMessage = useCallback(async (schedulingInfoParam?: SchedulingInfo) => {
+        const schedulingInfo = (schedulingInfoParam && 'scheduled_at' in schedulingInfoParam) ? schedulingInfoParam : undefined;
+
         if (persistentNotificationsEnabled) {
-            persistentNotificationsConfirmation(serverUrl, value, mentionsList, intl, sendMessage, persistentNotificationMaxRecipients, persistentNotificationInterval, currentUserId, channelName, channelType);
-        } else {
-            sendMessage();
+            const sendMessageWithScheduledPost = () => sendMessage(schedulingInfo);
+            await persistentNotificationsConfirmation(serverUrl, value, mentionsList, intl, sendMessageWithScheduledPost, persistentNotificationMaxRecipients, persistentNotificationInterval, currentUserId, channelName, channelType);
+            return Promise.resolve();
```

Comment:
`return Promise.resolve();` sounds off, but not sure if there is any more elegant way to handle this.

Reply:
That doesn't work with ESlint due to the error - `ESLint: Async arrow function expected no return value. (consistent-return)`. So if I'm returning something at one line, and not returning anything explicitly in other place, ESLint throws the error. Thats why all the `return Promise.resolve()` s.

Reply:
Ah I see, because you return other Promises.

Reply:
yes, correct.